### PR TITLE
Add PoolHandle

### PR DESCRIPTION
**Pull request by kagikn**

### CODE REVIEW

File: source/scripting/PoolHandle.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			{
+				return (byte)(Value & 0x000000FF);
+			}
+		}
+
+		public uint Index
+		{
+			get
+			{
+				return Value & 0xFFFFFF00;
```
Comment (njames93) [inline source/scripting/PoolHandle.cs:37]: Doesnt this want right shifting by 8

**Comment by kagikn** [comment]
I believe all script handles store a counter and an index. I confirmed entity handles and blip handles do at least. I'm very sure that Rockstar use this handle system since GTA 3D Universe, so I'll make sure this.

**Comment by njames93** [comment]
Blips to me look like the low 16 bits are used as an index for an array of all blip addresses, then high 16 bits are then used to check if the blip is valid by comparing with the short at blipAddress + 8

**Comment by kagikn** [comment]
Well, I should analyse how blip handles work again.

Wait, you know how to get a blip address from its handle? RPH has that feature though.

**Comment by njames93** [comment]
I only just found out, but yeah its pretty simple. not tried to look into mapping items in the blip struct because there isnt really any point

**Comment by kagikn** [comment]
After all, I've realized that not all kind of handles store index in high 24 bits and counter in low 8 bits, so I'll close this PR.
By the way, if PoolObject.Exists is an extension method, it would be great. We could verify both that a instance isn't null and that its handle is valid, only with one method.

**Pull request closed without merging**